tests/conftest.py

```diff
@@ -917,6 +920,52 @@ def create_mock_mqtt(*args, **kwargs) -> MqttMockHAClient:
         yield _setup_mqtt_entry
 
 
+@pytest.fixture
+def yaml_config() -> ConfigType | None:
+    """Fixture to override configuration.yaml using mock_yaml_config.
+
+    Patches the content of configuration.yaml with the
+    yaml dump of a config dict.
```

Comment:
This docstring is misleading IMHO, since the fixture itself doesn't patch anything.

Instead, the docstring should explain that the fixture provides a default (empty) configuration, which can be replaced by parametrizing `yaml_config` in tests.

This is a similar fixture:
https://github.com/home-assistant/core/blob/21b9f1f1db638c65c9eec6b28d43c8f78a17431e/tests/conftest.py#L1029-L1036